# Weekly Todo list

* [Week 42](#week-42)
* [Week 43](#week-43)
* [Week 44](#week-44)
* [week 45](#week-45)
* [week 46](#week-46)

# **18.10.2021 - 22.10.2021** {#week-42}

## IDH PhD project
* Look at clinical data from Verhaak cohort & do OS IDHwt analysis

* Look at cell lines previous analysis

* Make a signature of response

* Make a poster

## Splicing project

* Look at rMATS code [**DONE**]

* Read rMATS paper [**DONE**]

* Learn how to use rMATS [**DONE**]

* Learn how to use rmatsashimiplot [**DONE**]

## Wet-lab

* Do a clear recap on upcoming experiments [**DONE**]

## Deconvolution

* Try to find a way to investigate on the 40 genes

## Biblio

* Make a summary of a paper [3/3]
  * Genome_wide_analysis_of_focal_DNA_hypermethylation_in_IDH_mutant_AML_samples [**DONE**]
  * The_implications_of_IDH_mutations_for_cancer_development_and_therapy [**AML part DONE**]
  * The STAT3-MYC Axis Promotes Survival of Leukemia Stem Cells by Regulating SLC1A5 and Oxidative Phosphorylation [**DONE**]

* Look at paper already found on RELA

* Read paper on MYC


## Scripting

* Create Github.io? [**DONE**]

* Remake a clear rmd file [**DONE**]


# **25.10.2021 - 29.10.2021** {#week-43}

## Communication

* Make a part of the Team 18 poster [**DONE**]

* Redesign pictures and text for Cancer cell symposium poster

* Prepare a flash talk for Cancer cell symposium

## IDH PhD project

* Look at clinical data from Verhaak cohort & do OS IDHwt analysis

* Continue to look at cell lines previous analysis [**DONE**]

* Make a signature of response

* Start wet-lab [**DONE**]

* Look at Koichi RNAseq if bam files available [**DONE**]

## Biblio

* Find 3 papers [4/3] [**DONE**]

* Make a summary of a paper [0/3]  
  * Metabolic reprogramming and epigenetic modifications on the path to cancer
  * Genetic perturbation of PU.1 binding and chromatin looping at neutrophil enhancers associates with autoimmune disease
  * Mapping long-range promoter contacts in human cells with high-resolution capture Hi-C
  * EXTRA already read: Rewiring of the Transcription Factor Network in Acute Myeloid Leukemia [**DONE**]

* Do a thesis introduction [**will never ends**]

* Look at CHO abstracts

* Do method and team method bibliography

## Scripting

* Create Github.io? [**DONE**]

* Remake a clear rmd file

# **02.11.2021 - 05.11.2021** {#week-44}

## IDH PhD Project

* Put tables of DEG, Mviper etc. on git .io

* Find datasets on glioma IDH

* Continue Introduction

## Communication

* Make poster for Cancer cell symposium

* Record a flash talk for Cancer cell symposium

## Deconvolution

* Redo Venn diagram and analyze results

## Wet lab

* Wednesday --> Subculture 1:30pm --> 2:30pm

* Thursday --> Make "ampoule de congÃ©lation" 4:00pm --> 6:00pm

* Friday --> test mycoplasme --> 1:30pm --> 2:30pm

## Bibliography

* Make a summary of a paper [0/3]  
  * Metabolic reprogramming and epigenetic modifications on the path to cancer
  * Genetic perturbation of PU.1 binding and chromatin looping at neutrophil enhancers associates with autoimmune disease
  * Mapping long-range promoter contacts in human cells with high-resolution capture Hi-C

# **08.11.2021 - 10.11.2021** {#week-45}

## Team 18 Informatic park

* Install computer for new person
* Buy some screens for the PC of the salle commune

## Deconvolution

* Redo Venn diagram and heatmap

## Communication

* CDRC poster presentation on Monday

## IDH PhD Project

* Put tables of DEG, Mviper etc. on git .io

* Find datasets on glioma IDH

* Continue Introduction

* Compare 56-11 to Molm14

## Wet lab

Monday --> Subculture (3 samples)  
Tuesday --> Subculture + Dox +/- AGI  
Wednesday --> Subculture (5 samples)  

## Conferences

* Cell symposium

# **15.11.2021 - 19.11.2021** {#week-46}

## Team 18 Informatics

* Buy some screens and stuff for the PC of the salle commune
* Test PC commun and find out what happen to Thomas's computer

## Deconvolution

* Redo figure 2 of the paper

## IDH PhD Project

* Put tables of DEG, Mviper etc. on git .io

* Redo DNA methylation analysis

* From differences between Molm and HL60, make a signature to compare Good versus bad responders.

* Find datasets on glioma IDH

* Continue Introduction

* Check gene expression variability through cell lines & Patient response

## Wet lab

* ATP
* Scenith
* Rhod2
* Subcultures

## Bibliography

* Find a paper on methods for analysis of DNA met and/or RNAseq data

* Make a summary of a paper [0/1]  
  * Non-genetic mechanisms of therapeutic resistance in cancer

## Formations

* Thursday --> Programme mentorat
