# Meeting notes

* [29.10.2021](#29-10-2021)


# 29.10.2021 {#29-10-2021}

Chez les patients NR vs R à l'IDHi : **RELA** **STAT3** HIF1a **MYC** [link](https://alexishucteau.github.io/PhD_project/Analysis/Patients_samples_Koichi_and_Verhaak_analysis#NR_vs_R)  
Chez les patients High OS vs Low OS : **STAT3** MAPK1 FYN CTNN [link](https://alexishucteau.github.io/PhD_project/Analysis/Patients_samples_Koichi_and_Verhaak_analysis#High_OS_vs_Low_OS10)  
Dans les lignées HL60 IDHm +/-IDHi : SP1 ESR1 CEBPa **RELA** **STAT3** [link](https://alexishucteau.github.io/PhD_project/Analysis/HL60_MOLM14_RNAseq_analysis#HL60_+-_IDHi)  
Dans les lignées MOLM14 IDHm +/-IDHi : **MYC** FOS CEBPa NFKB1 [link](https://alexishucteau.github.io/PhD_project/Analysis/HL60_MOLM14_RNAseq_analysis#MOLM14_+-_IDHi)  
Dans les lignées HL60 +/- IDHm : STAT1 FOS **STAT3** GATA3 CEBPA [link](https://alexishucteau.github.io/PhD_project/Analysis/HL60_MOLM14_RNAseq_analysis#HL60_+-_IDHm)   

3 Genes potentielles : RELA, STAT3 et MYC  

STAT3 linked to MYC in LSCs OXPHOS activity  

bam files of Koichi's available  
Need storage to manage bam files

## Summary of the meeting

* HL60 --> NR to IDHi?
* MOLM14 --> R to IDHi?  
* Model TF that distinguish NR from R
* Algo de prediction
* Look at Fig3 Jexpmed
* THP1 --> Transcriptomes?
* STAT3 --> Chimio resistance (connue)
* FLT3 --> Chromatine modifiée
* Data IDHi glioma?
