---
title: "HL60 & MOLM14 impact of IDHmutations and inhibition, RNAseq analysis"
author: Alexis Hucteau
date: "`r Sys.Date()`"
output:
    html_document:
      toc: yes
      toc_float: yes
      theme: united
editor_options:
  markdown:
    wrap: sentence
---

Back to [PhD Project](https://alexishucteau.github.io/PhD_project/)

[Source code](https://alexishucteau.github.io/PhD_project/Analysis/source_code/HL60_MOLM14_RNAseq_code.R)

[Notes on cell lines](https://alexishucteau.github.io/PhD_project/Analysis/In_silico/Cell_lines_notes)

[Patients samples analysis](https://alexishucteau.github.io/PhD_project/Analysis/Patients_samples_Koichi_and_Verhaak_analysis)

```{r, include = F}
library(hugene20sttranscriptcluster.db)
library(oligo)
library(dplyr)
library(FactoMineR)
library(factoextra)
library(ggsignif)
library(viper)
library(dorothea)
library(data.table)
library(stringr)
library(igraph)
library(ggplot2)

load("~/DATA/Cell_lines_analysis.RData")
```


# **Analysis of HL60 MOLM14 IDHm cell lines +/- IDHi**

## PCA analysis

```{r}
plot.PCA(PCA_analysis)
fviz_eig(PCA_analysis, addlabels = T)
```

## Variability of gene expression {#var}

```{r}
ggplot(Variability_in_cell_lines, aes(x= Pheno, y = Variability, fill = Pheno))+
  geom_boxplot()+
  geom_point(size = 0.5)+
  xlab("Phenotypes")+
  geom_signif(comparisons = list(c("HL60.Mut.AGI5198", "HL60.Mut.DMF"), c("MOLM14.Mut.AGI5198", "MOLM14.Mut.DMF")))+
  labs(title = "Gene expression variability depending OS")
```

## TF analysis

### HL60_Mut_IDHi_vs_no_treat

```{r HL60_Mut_IDHi_vs_no_treat}
plot(HL60_Mut_IDHi_vs_no_treat$mrs)
```

### HL60_Mut_vs_HL60_WT

```{r HL60_Mut_vs_HL60_WT}
plot(HL60_Mut_vs_HL60_WT$mrs)
```

### MOLM14_Mut_IDHi_vs_no_treat

```{r MOLM14_Mut_IDHi_vs_no_treat}
plot(MOLM14_Mut_IDHi_vs_no_treat$mrs)
```

## Eigen centrality analysis

### HL60 +/- IDHi

```{r}
Do_cool_scatterplot(HL60_Mut_IDHi_vs_no_treat_network$features, "HL60 +/- IDHi")
```

### HL60 +/- IDHm

```{r}
Do_cool_scatterplot(HL60_Mut_vs_HL60_WT_network$features, "HL60 +/- IDHm")
```

### MOLM14 +/- IDHi

```{r}
Do_cool_scatterplot(MOLM14_Mut_IDHi_vs_no_treat_network$features, "MOLM14 +/- IDHi")
```


## RELA and MYC expression in cell lines

### RELA in HL60 +/- IDHi Expression

```{r RELA in HL60 +/- IDHi Expression}
Make_gene_expr_boxplots(Transcriptomes, ID_RELA, "RELA", "HL60.Mut.AGI5198", "HL60_IDHi", "HL60.Mut.DMF", "HL60_DMF", Phenotype)
```

### MYC in HL60 +/- IDHi  Expression

```{r MYC in HL60 +/- IDHi  Expression}
Make_gene_expr_boxplots(Transcriptomes, ID_MYC, "MYC", "HL60.Mut.AGI5198", "HL60_IDHi", "HL60.Mut.DMF", "HL60_DMF", Phenotype)
```

### RELA in HL60 +/- IDHm  Expression

```{r RELA in HL60 +/- IDHm  Expression}
Make_gene_expr_boxplots(Transcriptomes, ID_RELA, "RELA", "HL60.Mut.None", "HL60_Mut", "HL60.WT.None", "HL60_WT", Phenotype)
```

### MYC in HL60 +/- IDHm  Expression

```{r MYC in HL60 +/- IDHm  Expression}
Make_gene_expr_boxplots(Transcriptomes, ID_MYC, "MYC", "HL60.Mut.None", "HL60_Mut", "HL60.WT.None", "HL60_WT", Phenotype)
```

### RELA in MOLM14 +/- IDHi  Expression

```{r RELA in MOLM14 +/- IDHi  Expression}
Make_gene_expr_boxplots(Transcriptomes, ID_RELA, "RELA", "MOLM14.Mut.AGI5198", "MOLM14_IDHi", "MOLM14.Mut.DMF", "MOLM14_DMF", Phenotype)
```

### MYC in MOLM14 +/- IDHi  Expression

```{r MYC in MOLM14 +/- IDHi  Expression}
Make_gene_expr_boxplots(Transcriptomes, ID_MYC, "MYC", "MOLM14.Mut.AGI5198", "MOLM14_IDHi", "MOLM14.Mut.DMF", "MOLM14_DMF", Phenotype)
```

## Network signatures

### HL60 IDHm + IDHi vs HL60 IDHm + DMF

![HL60 IDHi](../Pictures/HL60_IDHi.png)

### HL60 IDHm vs HL60 IDHwt

![HL60 m/wt](../Pictures/HL60_Mut.png)

### MOLM14 IDHm + IDHi vs MOLM14 IDHm + DMF

![MOLM14 IDHi](../Pictures/MOLM14_IDHi.png)

# **Is HL60 a Bad responder and M14 a good responder to IDHi?**

## TF analysis

### HL60 IDHm vs MOLM14 +DOX (IDHm)

```{r}
plot(HL60_m_vs_M14_m$mrs)
```

### HL60 IDHm + AGI vs MOLM14 +DOX (IDHm) +AGI

```{r}
plot(HL60_m_agi_vs_M14_m_agi$mrs)
```

## Eigen centrality analysis

### HL60 IDHm vs M14 +Dox (IDHm)

```{r}
Do_cool_scatterplot(HL60_m_vs_M14_m_network$features, "HL60 IDHm vs M14 IDHm")
```

### HL60 IDHm +AGI vs M14 +Dox +AGI

```{r}
Do_cool_scatterplot(HL60_m_agi_vs_M14_m_agi_network$features, "HL60 IDHi vs M14 IDHi")
```



## RELA and MYC expression in cell lines

### RELA in HL60 IDHm vs M14 IDHm Expression

```{r RELA in HL60 IDHm vs M14 IDHm Expression}
Make_gene_expr_boxplots(Transcriptomes, ID_RELA, "RELA", "HL60.Mut.DMF", "HL60_IDHm", "MOLM14.Mut.DMF", "M14_DMF", Phenotype)
```

### MYC in HL60 IDHm vs M14 IDHm  Expression

```{r MYC in HL60 IDHm vs M14 IDHm  Expression}
Make_gene_expr_boxplots(Transcriptomes, ID_MYC, "MYC", "HL60.Mut.DMF", "HL60_IDHm", "MOLM14.Mut.DMF", "M14_DMF", Phenotype)
```

### RELA in HL60 IDHm +AGI vs M14 IDHm +AGI  Expression

```{r RELA in HL60 IDHm +AGI vs M14 IDHm +AGI  Expression}
Make_gene_expr_boxplots(Transcriptomes, ID_RELA, "RELA", "HL60.Mut.AGI5198", "HL60_IDHi", "MOLM14.Mut.AGI5198", "M14_IDHi", Phenotype)
```

### MYC in HL60 IDHm +AGI vs M14 IDHm +AGI  Expression

```{r MYC in HL60 IDHm +AGI vs M14 IDHm +AGI  Expression}
Make_gene_expr_boxplots(Transcriptomes, ID_MYC, "MYC", "HL60.Mut.AGI5198", "HL60_IDHi", "MOLM14.Mut.AGI5198", "M14_IDHi", Phenotype)
```

## Network signatures

### HL60 IDHm vs M14 IDHm

<!-- ![HL60 IDHi](../Pictures/HL60_M14_IDHm.png) -->

### HL60 IDHm +AGI vs M14 IDHm +AGI

<!-- ![HL60 m/wt](../Pictures/HL60_M14_IDHi.png) -->