---
title: "Patients samples Koichi and Verhaak, RNAseq and affy analysis"
author: Alexis Hucteau
date: "`r Sys.Date()`"
output:
    html_document:
      toc: yes
      toc_float: yes
      theme: united
editor_options:
  markdown:
    wrap: sentence
---

Back to [PhD Project](https://alexishucteau.github.io/PhD_project/)

[Cell lines analysis](https://alexishucteau.github.io/PhD_project/Analysis/HL60_MOLM14_RNAseq_analysis)

[Source code](https://alexishucteau.github.io/PhD_project/Analysis/source_code/Patients_samples_Koichi_analysis_code.R)

[Source code](https://alexishucteau.github.io/PhD_project/Analysis/source_code/Patients_samples_Verhaak_analysis_code.R)


```{r, include = F}
library(FactoMineR)
library(factoextra)
library(viper)
library(dplyr)
library(ggsignif)

load("~/DATA/Patients_data2.RData")

Do_cool_scatterplot <- function(Feature, title){
  Feature <- dplyr::filter(Feature, Eigen_centrality > 0.0005 & Page_rank != 0 & ((P.Value < 0.05 & abs(logFC) > 1.5) | pval < 0.1))
  DEG <- ifelse(Feature$logFC < 0, "DOWN", "UP")
  ggplot(Feature, aes(x = log(Page_rank), y = log(Eigen_centrality), label = Gene, colour = DEG))+
    geom_text(check_overlap = T, size = 2, nudge_x = 0.05, hjust = 0, outlier.size = 0)+
    geom_point(size = 0.5)+
    labs(title = paste0("Network-based node prioritization ", title))+
    xlab("Page Rank (log)")+
    ylab("Eigen Centrality (log)")+
    scale_colour_manual(values=c("#0000FF", "#FF0000"))
}

Do_cool_scatterplot_Ver <- function(Feature, title){
  Feature <- dplyr::filter(Feature, Eigen_centrality > 0.0005 & Page_rank != 0 & (P.Value < 0.05 | pval < 0.1))
  DEG <- ifelse(Feature$logFC < 0, "DOWN", "UP")
  ggplot(Feature, aes(x = log(Page_rank), y = log(Eigen_centrality), label = Gene, colour = DEG))+
    geom_text(check_overlap = T, size = 2, nudge_x = 0.05, hjust = 0, outlier.size = 0)+
    geom_point(size = 0.5)+
    labs(title = paste0("Network-based node prioritization ", title))+
    xlab("Page Rank (log)")+
    ylab("Eigen Centrality (log)")+
    scale_colour_manual(values=c("#0000FF", "#FF0000"))
}
```

# Summary

* [PCA analysis](#pca)
  * [Koichi](#pca-koichi)
  * [Verhaak](#pca-verhaak)
* [Variability of gene expression](#var)
  * [Koichi](#var-koichi)
  * [Verhaak](#var-verhaak)
* [Transcription factor analysis](#tf)
  * [Koichi](#tf-koichi)
  * [Verhaak](#tf-verhaak)
* [Eigenvalue cantrality analysis](#eigen)
  * [Koichi](#eigen-koichi)
  * [Verhaak](#eigen-verhaak)
* [RELA and MYC expressions](#expressions)
  * [RELA IDHi](#rela-idhi)
  * [MYC IDHi](#myc-idhi)
  * [RELA OS](#rela-os)
  * [MYC OS](#myc-os)
* [Networks](#networks)
  * [IDHi](#network-idhi)
  * [OS](#network-os)
  
  
# PCA analysis {#pca}

## Koichi's dataset {#pca-koichi}

```{r PCA plots}
plot.PCA(PCA_rnaseq)
fviz_eig(PCA_rnaseq, addlabels = T)
```

## Verhaak's dataset {#pca-verhaak}

```{r}
plot.PCA(PCA_Verhaak)
fviz_eig(PCA_Verhaak, addlabels = T)
```

# Variability of gene expression {#var}

## Koichi's cohort {#var-koichi}

```{r}
ggplot(Variability_in_Patients, aes(x= Pheno, y = Variability, fill = Pheno))+
  geom_boxplot()+
  geom_point(size = 0.5)+
  xlab("Phenotypes")+
  geom_signif(comparisons = list(c("NR.B", "R.B"), c("R.B", "R.REL")))+
  labs(title = "Gene expression variability depending IDHi response")
```

## Verhaak Cohort {#var-verhaak}

```{r}
ggplot(Variability_in_Patients_Verhaak, aes(x= Pheno, y = Variability, fill = Pheno))+
  geom_boxplot()+
  geom_point(size = 0.5)+
  xlab("Phenotypes")+
  geom_signif(comparisons = list(c("High_OS", "Low_OS")))+
  labs(title = "Gene expression variability depending overall survival")
```


# TF analysis {#tf}

## NR vs R IDHi {#tf-koichi}

```{r TF mviper plots}
plot(NR_R_msviper$mrs)
```



## High OS vs Low OS {#tf-verhaak}

```{r}
plot(High_OS_vs_Low_OS_msviper$mrs)
```

# Eigen centrality analysis {#eigen}

## NR vs R {#eigen-koichi}

```{r}
Do_cool_scatterplot(NR_R_network$features, "NR vs R")
```

## High OS vs Low OS {#eigen-verhaak}

```{r}
Do_cool_scatterplot_Ver(High_OS_vs_Low_OS_network$features, "High OS versus Low OS")
```

# RELA and MYC expression Patients samples {#expressions}

## RELA in NR vs R Expression {#rela-idhi}

```{r}
Make_gene_expr_boxplots(RNAseq, "RELA", "NR.B", "Bad responder", "R.B", "Good responder", Factor_R_OR_NR_B)
```

## MYC in NR vs R Expression {#myc-idhi}

```{r}
Make_gene_expr_boxplots(RNAseq, "MYC", "NR.B", "Bad responder", "R.B", "Good responder", Factor_R_OR_NR_B)
```

## RELA in High OS vs Low OS {#rela-os}

```{r}
Make_gene_expr_boxplots(Verhaak_norm_data_SYMBOL, "RELA", "High_OS", "High_OS", "Low_OS", "Low_OS", OS)
```

## MYC in High OS vs Low OS {#myc-os}

```{r}
Make_gene_expr_boxplots(Verhaak_norm_data_SYMBOL, "MYC", "High_OS", "High_OS", "Low_OS", "Low_OS", OS)
```


# Network signatures {#networks}

## NR vs R {#network-idhi}

![NR vs R](../Pictures/NR_R.png)




## High OS vs Low OS {#network-os}

![High OS vs Low OS](../Pictures/High_low_OS.png)

```{r}
png("./ASNS expression Response to IDHi.png")
Make_gene_expr_boxplots(RNAseq, "ASNS", "NR.B", "Bad responder", "R.B", "Good responder", Factor_R_OR_NR_B)
dev.off()
```



```{r}
Pheno_chas <- lapply(levels(Factor_R_OR_NR_B), function(pheno){
  ref <- Factor_R_OR_NR_B == pheno
  chas_for_selected_pheno(pheno = ref, title = pheno, cells = all_cell_lines)
})
names(Pheno_chas) <- levels(Factor_R_OR_NR_B)
```

